### PR TITLE
refactor: move payments-example plugin to npm package

### LINKED ISSUE #6129: move payments-example plugin to its own npm package, and install npm version Reaction
New repo: https://github.com/reactioncommerce/plugin-payments-example

Convert the payments-example plugin. This includes moving all related files into a new repository, releasing that repo as an `npm` package, and then updating this repo, `reaction`, to use the `npm` version of the plugin.

Part of [this epic](https://github.com/reactioncommerce/reaction/issues/6107) to move all plugins into their own repositories and release as `npm` packages.

### PR DESCRIPTION
**Pull request by trojanh**
Resolves https://github.com/reactioncommerce/reaction/issues/6129
Impact: **minor**
Type: **refactor**

## Issue
Convert the payments-example plugin in npm package. This includes moving all related files into a new repository, releasing that repo as an npm package, and then updating this repo, reaction, to use the npm version of the plugin.

plugin => https://github.com/reactioncommerce/plugin-payments-example/pull/1

## Testing
Check if all the `payments-example` plugin api are visible and working like before

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -23,3 +23,4 @@ services:
 volumes:
   mongo-db4:
   api_node_modules:
+    - /Users/kiprosh/projects/reaction-payments-example/plugin-payments-example
```
Comment (kieckhafer) [inline docker-compose.dev.yml]: This should not be committed.

File: jest.config.cjs [hunk 2]
```diff
@@ -2,6 +2,16 @@
 // For a detailed explanation regarding each configuration property, visit:
 // https://jestjs.io/docs/en/configuration.html
 
+// Any packages that are published only as ESM need to be listed here
+const externalNodeModules.join("|") = [
```
Comment (kieckhafer) [inline jest.config.cjs]: 👍 Nice, been trying to think of the best way to lay this out so it doesn't get so crowded.

File: src/registerPlugins.js [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import registerAuthenticationPlugin from "@reactioncommerce/plugin-authentication";
 import registerSimpleAuthorizationPlugin from "@reactioncommerce/plugin-simple-authorization";
 import registerSystemInformationPlugin from "@reactioncommerce/plugin-system-information";
+import registerExamplePaymentsPlugin from "@reactioncommerce/plugin-payments-example";
```
Comment (kieckhafer) [inline src/registerPlugins.js]: Please alphabetize this list.

**Review by kieckhafer — CHANGES_REQUESTED** [review]

File: docker-compose.dev.yml (hunk 1 above)
Comment (trojanh) [reply docker-compose.dev.yml]: yeah, sorry I did this by mistake.

**Comment by trojanh** [comment]
Thanks for reviewing @kieckhafer , I have made the suggested changes, please review it when you get a chance.

**Comment by trojanh** [comment]
@kieckhafer just noticed that jest node modules need to alphabetized as well, so pushing that change now

**Comment by kieckhafer** [comment]
@trojanh Thanks! I was about to comment on that, glad you caught it.

**Comment by trojanh** [comment]
sure @kieckhafer , fixed that now

**Comment by kieckhafer** [comment]
@trojanh now that your other package is published, can you run `npm install` locally, and then push the new `package-lock.js` that is created. I think that'll solve the CI issues.

File: jest.config.cjs [hunk 4]
```diff
@@ -176,7 +186,7 @@ const jestConfig = {
   // An array of regexp pattern strings that are matched against all source file paths, matched files will skip transformation
   transformIgnorePatterns: [
     // Any packages that are published only as ESM need to be listed here
-    "node_modules/(?!(@reactioncommerce/api-utils|@reactioncommerce/db-version-check|@reactioncommerce/plugin-simple-authorization|@reactioncommerce/plugin-system-information|@reactioncommerce/plugin-authentication)/)"
+    `node_modules/(?!(${externalNodeModules.join("|")}/)`
```
Comment (kieckhafer) [inline jest.config.cjs]: You need to add a parens to the end of this, that should fix the test issues.

```suggestion
    `node_modules/(?!(${externalNodeModules.join("|")})/)`
```

**Review by kieckhafer — CHANGES_REQUESTED** [review]

**Review by kieckhafer — APPROVED** [review]

**Comment by kieckhafer** [comment]
This is ready to go once https://github.com/reactioncommerce/plugin-payments-example has been made public.

**Pull request merged**